### PR TITLE
Show image upload errors

### PR DESCRIPTION
**Pull request by torchiaf**
#### PR Checklist
- Is this a multi-tenancy feature/bug?
    - [ ] Yes, the relevant RBAC changes are at:
- Do we need to backport changes to the [old Rancher UI](https://github.com/rancher/u), such as RKE1?
    - [ ] Yes, the relevant PR is at:
- Are backend engineers aware of UI changes?
    - [ ] Yes, the backend owner is:

Fixes https://github.com/harvester/harvester/issues/5014

### Technical notes summary

In case of Http errors during image uploads, the UI shows the error in images table.

### Screenshot/Video

Screenshot from 2024-01-23 18-35-24

### CODE REVIEW

**Comment by bk201** [comment]
Thanks for the fix. Does the user get the full error message if he or she hovers the mouse to "Error 403..."?

**Comment by torchiaf** [comment]
> Thanks for the fix. Does the user get the full error message if he or she hovers the mouse to "Error 403..."?

Full message will be displayed. "Error 403..." is just a hard-coded value in my test.

File: pkg/harvester/models/harvesterhci.io.virtualmachineimage.js [hunk 1]
```diff
@@ -105,6 +105,10 @@ export default class HciVmImage extends HarvesterResource {
       if (this.spec.sourceType === 'download') {
         return 'Downloading';
       } else if (this.spec.sourceType === 'upload') {
+        if (this.uploadError) {
+          return 'Failed';
+        }
+
```
Comment (Yu-Jack) [inline pkg/harvester/models/harvesterhci.io.virtualmachineimage.js:111]: Nit: Is it better to remove else here for reading?

```js
if (imported?.status === 'Unknown') {
  if (this.spec.sourceType === 'download') {
    return 'Downloading';
  } 
  
  if (this.spec.sourceType === 'upload') {
    if (this.uploadError) {
      return 'Failed';
    }
    return 'Uploading';
  }
  
  return 'Exporting'
}
```

Comment (torchiaf) [reply pkg/harvester/models/harvesterhci.io.virtualmachineimage.js:111]: Correct, thanks!

**Review by bk201 — APPROVED** [review]

**Comment by torchiaf** [comment]
Reopen for QA tests.

**Comment by torchiaf** [comment]
@mergify backport release-harvester-v1.2

**Pull request merged**